src/Bicep.LangServer/Providers/BicepCompilationProvider.cs

```diff
@@ -13,14 +15,17 @@ namespace Bicep.LanguageServer.Providers
     /// <remarks>This class exists only so we can mock fatal exceptions in tests.</remarks>
     public class BicepCompilationProvider: ICompilationProvider
     {
+        // we don't want to recreate this every time we compile a file, because creation is expensive
+        private ResourceTypeRegistrar resourceTypeRegistrar = new ResourceTypeRegistrar(new AzResourceTypeProvider());
```

Comment:
>ResourceTypeRegistrar [](start = 66, length = 21)

nit - Since we have DI in the language server, it should come in through that via the constructor. (You can register a manually created instance or can register the type provider in the container - either works. Also fine if it comes in on another PR.)